src/Features/Core/Portable/EditAndContinue/EditSession.cs

```diff
@@ -251,110 +237,139 @@ private LinePositionSpan GetUpToDateSpan(ActiveStatementDebugInfo activeStatemen
+                    return _lazyBaseActiveExceptionRegions;
+                }
+
                 var baseActiveStatements = await BaseActiveStatements.GetValueAsync(cancellationToken).ConfigureAwait(false);
                 var instructionMap = baseActiveStatements.InstructionMap;
                 using var builderDisposer = ArrayBuilder<ActiveStatementExceptionRegions>.GetInstance(instructionMap.Count, out var builder);
                 builder.Count = instructionMap.Count;
 
+                bool hasOutOfSyncDocuments = false;
+
                 foreach (var activeStatement in instructionMap.Values)
                 {
-                    var document = BaseSolution.GetDocument(activeStatement.PrimaryDocumentId);
+                    bool isCovered;
+                    ImmutableArray<LinePositionSpan> exceptionRegions;
 
-                    var sourceText = await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
-                    var syntaxRoot = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+                    // Can't calculate exception regions for active statements in out-of-sync documents.
+                    var (document, _) = await DebuggingSession.LastCommittedSolution.GetDocumentAndStateAsync(activeStatement.PrimaryDocumentId, cancellationToken).ConfigureAwait(false);
+                    if (document != null)
+                    {
+                        var sourceText = await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
+                        var syntaxRoot = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
 
-                    var analyzer = document.Project.LanguageServices.GetService<IEditAndContinueAnalyzer>();
-                    var exceptionRegions = analyzer.GetExceptionRegions(sourceText, syntaxRoot, activeStatement.Span, activeStatement.IsNonLeaf, out var isCovered);
+                        var analyzer = document.Project.LanguageServices.GetService<IEditAndContinueAnalyzer>();
+                        exceptionRegions = analyzer.GetExceptionRegions(sourceText, syntaxRoot, activeStatement.Span, activeStatement.IsNonLeaf, out isCovered);
+                    }
+                    else
+                    {
+                        // Document is either out-of-sync, design-time-only or missing from the baseline.
+                        // If it's missing or design-time-only it can't have active statements.
+                        hasOutOfSyncDocuments = true;
+                        isCovered = false;
+                        exceptionRegions = default;
+                    }
 
                     builder[activeStatement.Ordinal] = new ActiveStatementExceptionRegions(exceptionRegions, isCovered);
                 }
 
-                return builder.ToImmutable();
+                var result = builder.ToImmutable();
+
+                // Only cache results if no active statements are in out-of-sync documents.
```

Comment:
What happens if the document later comes back in sync, since OutOfSync is (theoretically) transient?

Reply:
We do not cache the result until all documents that contain active statements are in-sync with debuggee. Until then the method returns a different result each time with blanks for out-of-sync docs. The callers handle not having exception handler for out-of-sync doc.